scripts/reborn_webui_v2_live_qa/test_run_live_qa.py

```
@@ -252,6 +259,340 @@ def fake_capability_run_statuses(_reborn_home, capability_ids):
+                )
+            )
+
+        self.assertFalse(result.success)
+        self.assertEqual(captured_prompts, ["original sheet prompt"])
+        self.assertEqual(result.details["trigger_records_after"], 0)
+        self.assertIn("did not add a trigger_record", result.details["error"])
+
+    def test_routine_creation_case_can_preinstall_extensions(self):
+        captured: dict[str, object] = {}
+
+        async def fake_live_chat_with_extensions_case(_ctx, **kwargs):
```

Ruff (0.15.18)

[warning] Missing return type annotation for private function `fake_live_chat_with_extensions_case`

(ANN202)